src/Casters/ArrayCaster.php

```diff
@@ -60,7 +60,7 @@ private function castItem(mixed $data)
         }
 
         if (is_array($data)) {
-            return new $this->itemType(...$data);
+            return call_user_func_array("{$this->itemType}::new", $data);
```

Comment:
We may have to check whether the item type is a Data Transfer Object, and then use the `new` method. That said, we may also want to use `newWithoutValidation` and then have to iterate through traversables to validate the DTOs.